From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chao zeng <chao.zeng@siemens.com>
Date: Thu, 4 May 2023 10:31:09 +0800
Subject: [PATCH] Dynamic add user button label

Not all platform has the user button supported, so dynamic
to add the user button label.

Signed-off-by: chao zeng <chao.zeng@siemens.com>
---
 hardware/intel/mraa-gpio-din.html | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hardware/intel/mraa-gpio-din.html b/hardware/intel/mraa-gpio-din.html
index df76b93..83ef75f 100644
--- a/hardware/intel/mraa-gpio-din.html
+++ b/hardware/intel/mraa-gpio-din.html
@@ -34,6 +34,9 @@
                 if (data === 7) { t = "Banana"; }
                 if (data === 26) { t = "IOT2050"; }
                 $('#type-tip').text(t);
+                if (data === 26) {
+                    $('#node-input-pin').append($("<option></option>").attr("value",20).text("USER button"));
+                }
                 $('#node-input-pin').val(pinnow);
             });
             $.getJSON('mraa-version/'+this.id,function(data) {
@@ -68,7 +71,6 @@
             <option value="17">D17</option>
             <option value="18">D18</option>
             <option value="19">D19</option>
-            <option value="20">USER button</option>
         </select>
     </div>
     <div class="form-row">
